tracer/src/Datadog.Trace/Agent/DiscoveryService/AgentConfiguration.cs

```diff
@@ -60,4 +62,6 @@ public AgentConfiguration(
     public bool ClientDropP0s { get; }
 
     public bool SpanMetaStructs { get; }
+
+    public bool SpanEvents { get; }
```

Comment:
The part of the implementation that I don't see yet (maybe you're already working on this locally) is propagating the bool value of whether native span events are supported and propagating that back to the `Tracer` or `TracerManager` instances. Take a look at `tracer\src\Datadog.Trace\Logging\TracerFlare\TracerFlareManager.cs` to see how it uses the `HandleConfigUpdate` method to respond to `AgentConfiguration` updates. I expect you'll need to do something similar